test/boost/cql_auth_syntax_test.cc

```diff
@@ -320,3 +320,7 @@ BOOST_AUTO_TEST_CASE(grant_role) {
 BOOST_AUTO_TEST_CASE(revoke_role) {
     test_valid("REVOKE soldier FROM boromir;");
 }
+
+BOOST_AUTO_TEST_CASE(create_role_with_salted_hash) {
+    test_valid("CREATE ROLE adam WITH SALTED HASH = 'something';");
```

Comment:
What happens if the salted hash doesn't match any hash algorithm (as in this example) and someone tries to login? I expect it will fail, but better to test it doesn't crash.

Reply:
That's a very good point. I'll add a test